test-data/unit/check-inference.test

```diff
@@ -3176,3 +3176,27 @@ reveal_type(f(lambda: 1))  # N: Revealed type is 'builtins.int*'
 def g() -> None: pass
 
 reveal_type(f(g))  # N: Revealed type is 'None'
+
+-- Tests for checking if annotation is needed
+-- --------------
+
+[case testexplicitAnnotationNeededForLists]
+x = [1, 'two', 3.5]  # E: Need type annotation for 'x' \
+                     # N: Suggested annotation: List[Any] or List[Union[float, int, str]]
+
+y = [2, 'three', object()]
```

Comment:
I wanted to add a similar test for ``set`` but I guess ``/fixtures/set.pyi`` does not supprt ``float``. So the test was failing. Is there a reason that this is missing from the set fixture?

Reply:
Not really, we tend to just put things in fixtures when we need them. Feel free to add it, but adding things to fixtures incurs a slight performance penalty, so if possible it may be better to switch to another fixture that does have float.

Or just use a different type than float.

Reply:
Oh okay! I wasn't sure about fixtures and hence the confusion. I'll redo the test without floats, it wouldn't make much of a difference. Thank you!